src/unix/linux_like/mod.rs

```diff
@@ -234,10 +235,95 @@ s! {
         pub machine: [c_char; 65],
         pub domainname: [c_char; 65],
     }
+
+    pub struct if_nameindex {
+        pub if_index: c_uint,
+        pub if_name: *mut c_char,
+    }
+
+    #[cfg_attr(
+        any(
+            target_pointer_width = "32",
+            target_arch = "x86_64",
+            target_arch = "powerpc64",
+            target_arch = "mips64",
+            target_arch = "mips64r6",
+            target_arch = "s390x",
+            target_arch = "sparc64",
+            target_arch = "aarch64",
+            target_arch = "riscv64",
+            target_arch = "riscv32",
+            target_arch = "loongarch64",
+            target_os = "emscripten"
+        ),
+        repr(align(4))
+    )]
+    #[cfg_attr(
+        not(any(
+            target_pointer_width = "32",
+            target_arch = "x86_64",
+            target_arch = "powerpc64",
+            target_arch = "mips64",
+            target_arch = "mips64r6",
+            target_arch = "s390x",
+            target_arch = "sparc64",
+            target_arch = "aarch64",
+            target_arch = "riscv64",
+            target_arch = "riscv32",
+            target_arch = "loongarch64",
+            target_os = "emscripten"
+        )),
+        repr(align(8))
+    )]
+    #[cfg(not(target_os = "android"))]
+    pub struct pthread_mutexattr_t {
+        #[doc(hidden)]
+        size: [u8; crate::__SIZEOF_PTHREAD_MUTEXATTR_T],
+    }
```

Comment:
For this PR, I'd like to avoid commonizing things that aren't already commonized, i.e. pulling them out of android/emscripten up to linux_like (thinking both about conflicts for the backport and changes that aren't simple). Could you restore changes to the android/emscripten files? This means some things will be a bit duplicated in the l4re module, but I'd rather clean it up later than dealing with >2 platforms here.

The `pthread_*` types stand out in particular since they have such a complex definition.